File: codes/openmp/my_OpenMP/eval_y.py
import numpy as np

# Parameters
n = 100_000
dx = 1.0 / n

# Create x array from 1*dx to n*dx
x = np.arange(1, n + 1) * dx

# Compute y = sin(x) * exp(-x^2)
y = np.sin(x) * np.exp(-x**2)

print(f"y array has been filled with n = {n}")

File: codes/openmp/my_OpenMP/eval_y.f90
program evaluate_y
	
	use omp_lib			! importing something in program should go first
	implicit none
	integer :: i
	integer :: n_threads
	integer, parameter :: n = 100000
	real(kind=8), dimension(n) :: y
	real(kind=8) :: x, dx
	
	dx = 1.d0 / n
	
	!$ n_threads = THREADS
	
	!$ call omp_set_num_threads(n_threads)
	
	print *, "using OpenMP with threads =", n_threads 
	!$omp parallel do private(x)
	do i = 1, n
		x = i * dx
		y(i) = sin(x) * exp(-x**2)
	enddo
	!$omp end parallel do
	
	print *, 'y array has been filled with n=', n
	
	! compile as:  gfortran -cpp -fopenmp -DTHREADS=4 eval_y.f90
end program evaluate_y
